TheMET/TheMET/TagListView/TagListView.swift

```diff
@@ -0,0 +1,194 @@
+//
+//  TagListView.swift
+//  TheMET
+//
+//  Created by Анна Ситникова on 17/05/2023.
+//
+
+import Foundation
+import UIKit
+
+typealias TagView = UIButton
+
+class TagListView: UIView {
+    
+    var tags: [String] = [] {
+        didSet {
+            self.updateTags()
+        }
+    }
+    var maxNumberOfLines: Int = 7 {
+        didSet {
+            self.updateTags()
+        }
+    }
+    
+    private let showMoreTagsButton = UIButton()
+    private var tagViews: [TagView] = []
+    
+    private let tagsXSpacing: CGFloat = 12
+    private let tagsYSpacing: CGFloat = 10
+    
+    private let offsetByX: CGFloat = 0
+    private let offsetByY: CGFloat = 0
+    
+    private var totalHeight: CGFloat = 0 {
+        didSet {
+            self.bounds = CGRect(origin: self.bounds.origin,
+                                 size: CGSize.init(width: self.bounds.width,
+                                                   height: self.totalHeight))
+        }
+    }
+    
+    override open var intrinsicContentSize: CGSize {
+        return CGSize(width: self.bounds.width, height: self.totalHeight)
+    }
+    
+    override var bounds: CGRect {
+        didSet {
+            guard self.bounds.width != oldValue.width else { return }
+            self.updateTags()
+        }
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.showMoreTagsButton.apply(radius: 0, backgroundColor: .clear , fontColor: UIColor(named: "blueberry"), font: NSLocalizedString("serif_font", comment: ""), fontSize: 14, buttonTitle: NSLocalizedString("button_title.show_more_tags_cta", comment: ""))
+        self.showMoreTagsButton.addTarget(self, action: #selector(showMoreButtonDidTap), for: .touchUpInside)
+        self.showMoreTagsButton.frame = CGRect(origin: .zero,
+                                               size: self.showMoreTagsButton.intrinsicContentSize)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func sizeThatFits(_ size: CGSize) -> CGSize {
```

Comment:
This method significantly duplicates the `layoutTagView` one. Let's split out a common part of these methods. This method should receive a `width` value to calculate next outcome:
- The frames of all tag-views
- The frame of the `shoreMoreButton` (optional)
- The result layout's `height` 

For simplicity all three values could be combined in a Tuple. (Please google and read about tuples in [this arcticle](https://abhimuralidharan.medium.com/tuple-in-swift-a9ddeb314c79) )

So the API of this method would look smth like:
```
func calculateLayout(for width: CGFloat) -> (tagViewFrames: [CGRect], showMorButtonFrame: CGRect?, height: CGFloat)
```

Then use the method in both `sizeThatFits` and `layoutTagViews`. First one could just use a `height`, while a second should use all the components of the result and setup `tagViews'` frames, `showMoreButton's` frame and `totalHeight`

Reply:
fixed